### PR TITLE
Fix delay due to keydown and add multiple key presses

### PR DESCRIPTION
**Pull request by ALai2**
I added a gameloop which updates the game state and canvas every 50 milliseconds. This removes the delay when updating the canvas every time a key is pressed. The key down event now only adds a key value pair to a keysPressed dictionary, which the gameloop function uses. The complementary key up event deletes the key value pair from the dictionary. This way of using the key down event allows for the user to press multiple keys at once to move the snake, allowing the snake to move diagonally without the user needing to quickly switch between keys. A mousedown event was also added so that the snake only moves when the user clicks on the canvas beforehand.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -63,34 +69,58 @@ function drawSnake() {
...
-    ctx.rect(x[z], y[z], SNAKE_SQUARE_SIZE, SNAKE_SQUARE_SIZE);
-    ctx.fillStyle = "#ff0000";
-    ctx.fill();
+
+    if (z == 0) {
+      ctx.fillStyle = "#000000";
+    } else if (z % 2 == 0) {
+      ctx.fillStyle = "#FF9900";
+    } else {
+      ctx.fillStyle = "#FF0000";
```
Comment (linxiz) [inline portfolio/src/main/webapp/script.js]: nit: lower case

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -100,7 +130,23 @@ window.onkeydown = function(event) {
...
+window.onkeydown = function(event) {
+  var values = getCanvasAndContext();
+  var canvas = values.canvas;
+  if (prevMousePressLoc == canvas) {
+    event.preventDefault();
   
-  // Drawing snake at new position
-  drawSnake();
+    var keyPr = event.keyCode; // Key code of key pressed
+    keysPressed[keyPr] = true;
```
Comment (linxiz) [inline portfolio/src/main/webapp/script.js]: Do you want to use a Set instead? the boolean value is never used.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -100,7 +130,23 @@ window.onkeydown = function(event) {
...
+
+window.onkeydown = function(event) {
+  var values = getCanvasAndContext();
+  var canvas = values.canvas;
+  if (prevMousePressLoc == canvas) {
+    event.preventDefault();
   
-  // Drawing snake at new position
-  drawSnake();
+    var keyPr = event.keyCode; // Key code of key pressed
```
Comment (linxiz) [inline portfolio/src/main/webapp/script.js]: nit: name the variable as keyCode to be more readable

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -63,34 +69,58 @@ function drawSnake() {
...
-  
+    // Drawing snake at new position
+    drawSnake();
+  }
+
+  setTimeout(gameLoop, 50);
+}
+
+// move snake inside the canvas using arrow keys
+function moveSnake(keyPr) {
```
Comment (linxiz) [inline portfolio/src/main/webapp/script.js:123]: add jsdoc for input param

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (ALai2) [reply portfolio/src/main/webapp/script.js]: I added a commit which replaces the dictionary with a set.

**Review by linxiz — APPROVED** [review]

**Pull request merged**